spec/controllers/submissions_controller_spec.rb

```diff
@@ -184,6 +184,59 @@
           expect(files['TestShapes.java']).to be_nil
         end
       end
+
+      context 'when creating a folder with required files' do
```

Comment:
Add some tests to make sure the folder actually get created or not (don't just check the http status)

Reply:
Right, of course. This did lead me to catch an edge case (not sure if it is known): 

When attempting to upload a subdirectory without initially creating its parent directory, it fails silently (that is, it returns a 200, but no folders are actually added to the repo).

I haven't fully diagnosed why this is happening, but in practice I'm fairly certain it never will. Using the web interface, only one directory can be created at a time. using zip files, all of the directories are created in order. Someone could make a post with those parameters, but it isn't supported. Either way, do we think this is at all worth investigating and either creating the parent directory if necessary or (more likely) catching this and returning an error?

Reply:
Can you add a test for that case where you check if it's displaying flash messages? I don't think we're necessarily changing the return code if a transaction fails (we probably should but we don't) but we should at least be writing the error messages to the flash hash.